Use the web design gem

jeremiak opened the pull request:
Requires the US Web Design Standards gem and refactors the CSS accordingly.

Commit:
Use web design standards gem

Commit:
Use web design standards gem

Commit:
Detail manifest view

Commit:
Detail manifest view

Commit:
Refactor CSS using USWDS

Commit:
Refactor CSS using USWDS

Commit:
Merge in master

Commit:
Include Sass partials

Commit:
Include Sass partials

Commit:
Use semantic markup to simply HTML and CSS

Instead of having lots of nested divs with
special classes, let's use the <dl> element
to display the information about a manifest.

This reduces the markup required to be in HTML,
and thus reduces the complexity of the CSS.

Commit:
Use semantic markup to simply HTML and CSS

Instead of having lots of nested divs with
special classes, let's use the <dl> element
to display the information about a manifest.

This reduces the markup required to be in HTML,
and thus reduces the complexity of the CSS.

Commit:
Remove bourbon, neat, etc

Now that we have the USWDS gem installed, we don't need to worry
about these other libraries (they form the basis of the USWDS).

Commit:
Remove bourbon, neat, etc

Now that we have the USWDS gem installed, we don't need to worry
about these other libraries (they form the basis of the USWDS).

jessieay commented on config/routes.rb:
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
   root to: 'manifests#index'
-  resources :manifests, only: [:new, :create]
+  resources :manifests, only: [:new, :show]
```
we need to add back in `create` here so the specs don't fail :) (should be `[:new, :create, :show]`)

jessieay commented on app/views/layouts/application.html.erb:
```diff
@@ -13,19 +13,30 @@
+      </div>
+      <div class="usa-width-one-half text-right">
         <%= link_to 'Home', root_path %>
-        <%= link_to 'Industry', new_submission_path %>
```
do we still want to link to the new submission path in the header somewhere?

jessieay commented on app/views/manifests/index.html.erb:
```diff
@@ -1,7 +1,175 @@
+      <input id="generator_name" ng-model="data.generator.name" name="generator_name" type="text" class="ng-pristine ng-valid ng-touched">
+
+      <label class="featured" for="tsdf_name">Disposal facility name</label>
+      <input id="tsdf_name" ng-model="data.tsdf_name" name="tsdf_name" type="text" class="ng-pristine ng-untouched ng-valid">
```
`ng` classes left in from angular days?

jessieay commented on app/views/manifests/index.html.erb:
```diff
@@ -1,7 +1,175 @@
+          <label for="origin_state">Generator state</label>
+          <select name="origin_state" id="origin_state" class="usa-input-grid-tiny tiny-input">
+            <option value="">--</option>
+            <option value="AL">AL</option>
```
instead of listing all of these here, we can take advantage of https://github.com/18F/e-manifest/blob/master/app/helpers/states_helper.rb

``` ruby
<% us_states.each do |state| %>
 <option value=<%=state[1]%>><%=state[1]%></option>
<% end%$>
```

```
```

jessieay replied:
or better yet, in that same helper file we can define a method and use it:

``` ruby
# app/helpers/states_helper.rb

def state_option(state)
  "<option value=#{state}>#{state}</option>"
end

```

and then in here we do:

``` ruby
<% us_states.each do |state| %>
  <%= raw state_option(state[1]) %>
<% end%$>
```

jessieay commented on app/views/manifests/show.html.erb:
```diff
@@ -0,0 +1,89 @@
+    <dd><%= @manifest.tracking_number %></dd>
+  </dl>
+
+  <h3>Transporter</h3>
```
there could be any number of transporters. not sure if you'd rather deal with it now or later, but we should make sure this layout deals with that nicely

Commit:
More Angular clean up

Removing more angular classes that @jessieay found and using the states
option helper to generate the selects

Commit:
Make show method public

Commit:
Show real data for manifest

Commit:
Merge remote branch

Commit:
Change about to new submission path

Commit:
Use USWDS button classes

Commit:
Merge branch 'master' of github.com:18F/e-manifest into jk-us-web-design-gem
